pkg/controller/external_nofork.go

```diff
@@ -267,44 +266,12 @@ func (c *NoForkConnector) Connect(ctx context.Context, mg xpresource.Managed) (m
 	}, nil
 }
 
-func deleteInstanceDiffAttribute(instanceDiff *tf.InstanceDiff, paramKey string) error {
-	delete(instanceDiff.Attributes, paramKey)
-
-	keyComponents := strings.Split(paramKey, ".")
-	if len(keyComponents) < 2 {
-		return nil
-	}
-
-	keyComponents[len(keyComponents)-1] = "%"
```

Comment:
So does it mean Terraform does not actually care about the length diffs if the remaning `ResourceAttrDiff` are properly calculated?

Reply:
Yes, for maps and sets. No, for lists. Indicated in comments.